hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionFileSystem.java

```diff
@@ -189,8 +189,14 @@ Path createStoreDir(final String familyName) throws IOException {
    * @param familyName The name of column family.
    * @param policyName The name of the storage policy
    */
-  public void setStoragePolicy(String familyName, String policyName) {
-    FSUtils.setStoragePolicy(this.fs, getStoreDir(familyName), policyName);
+  public void setStoragePolicy(String familyName, String policyName) throws IOException {
+    if (this.fs instanceof HFileSystem) {
```

Comment:
could you please explain why handling HFileSystem separately here?

Reply:
I don't think it's a good idea adding dependency here to a specific FS implementation class, with the solely aim to reduce verbosity on TRACE logging (as the logging on `CommonFSUtils` are all TRACE). I understand, though, that the extra messaging can cause confusion while troubleshooting, so maybe we could specify the FS instance passed to `setStoragePolicy`along with the messages?

Reply:
@wchevreuil  We do the similar thing at several places in this file, it's not just the duplicate traces but to avoid the duplicate useless function calls. I believe title of the jira should be modified. 
Here is an example where we get the backing file system is case it is instance of hfilesystem

https://github.com/apache/hbase/blob/branch-1/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionFileSystem.java#L535

Reply:
> it's not just the duplicate traces but to avoid the duplicate useless function calls.

Well, it's a tradeoff indeed. I would still favour code isolation from specific impl logic, over the single additional call. Here we are making the code a bit tighter, with one more place to be changed in the event of a change at _HFileSystem_, for example.  If we agree to get this merged though, can we at least make sure we have tests for the two _if_ conditions now?